src/optimizer_interface.jl

```diff
@@ -119,6 +119,9 @@ function set_optimizer(
     # using the existing `model_cache`.
     model.moi_backend =
         MOI.Utilities.CachingOptimizer(backend(model).model_cache, optimizer)
```

Comment:
The copy of the optimizer attributes should be done by the `CachingOptimizer` constructor. It is checking that the `optimizer` is empty to make sure that it is in sync:
https://github.com/jump-dev/MathOptInterface.jl/blob/5e4cd36b5477d7f2a241edfed1e186848992bd71/src/Utilities/cachingoptimizer.jl#L97
but since `attach_optimizer` does not copy the optimizer attributes, then the constructor should copy them otherwise they are not in sync.